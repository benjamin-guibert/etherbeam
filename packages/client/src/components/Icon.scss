@use '../styles/resources/sizes';

.my-icon {
  display: inline-flex;
  align-items: center;

  &-label {
    padding-left: sizes.$spacer-s;
  }
}
