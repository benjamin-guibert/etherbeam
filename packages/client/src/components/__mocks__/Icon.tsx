import React, { ReactElement } from 'react'

const Icon = (): ReactElement => {
  return <span className="my-icon"></span>
}

export default Icon
