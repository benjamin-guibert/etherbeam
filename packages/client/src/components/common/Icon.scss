@import 'styles/variables';

.app-icon {
  &-xs {
    width: $app-size-xs;
    height: $app-size-xs;
  }

  &-sm {
    width: $app-size-sm;
    height: $app-size-sm;
  }

  &-m {
    width: $app-size-m;
    height: $app-size-m;
  }

  &-lg {
    width: $app-size-lg;
    height: $app-size-lg;
    vertical-align: -25%;
  }

  &-max {
    width: 100%;
    height: 100%;
  }

  @for $i from 2 through 10 {
    &-#{$i} {
      width: #{$i}em;
      height: #{$i}em;
    }
  }
}
