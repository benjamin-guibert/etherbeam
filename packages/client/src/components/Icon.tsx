import React, { FC } from 'react'
import { IconProp } from '@fortawesome/fontawesome-svg-core'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import './Icon.scss'

type LabelBreakpointProp = 'xs' | 's' | 'm' | 'l'
type ColorProp = 'dark' | 'positive' | 'negative'
type SizeProp = 'm' | 'l' | 'xl'

interface IconProps {
  icon: IconProp
  label?: string
  labelBreakpoint?: LabelBreakpointProp
  size?: SizeProp
  iconColor?: ColorProp
  iconSize?: SizeProp
  className?: string
}

const Icon: FC<IconProps> = ({ icon, label, labelBreakpoint, size, iconColor, iconSize, className }) => {
  const classNames = ['my-icon', size ? `my-size-${size}` : '', className].join(' ')
  const iconClassName = [iconColor ? `my-icon-${iconColor}` : '', iconSize ? `my-size-${iconSize}` : ''].join(' ')
  const labelClassName = ['my-icon-label', labelBreakpoint ? `my-d-min-${labelBreakpoint}` : ''].join(' ')

  return (
    <span className={classNames}>
      <span className={iconClassName}>
        <FontAwesomeIcon icon={icon} />
      </span>
      {!!label && <span className={labelClassName}>{label}</span>}
    </span>
  )
}

export default Icon
