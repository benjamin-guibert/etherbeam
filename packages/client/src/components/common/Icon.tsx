import React, { ReactElement } from 'react'
import './Icon.scss'

interface IconProps {
  src: unknown
  alt?: string
  size?: 'xs' | 'sm' | 'm' | 'lg' | 'xl' | number
  className?: string
  noVerticalAlignment?: boolean
}

const Icon = ({ src, alt = '', size = 'm', className = '', noVerticalAlignment }: IconProps): ReactElement => {
  const getClasses = () => {
    const classes = ['app-icon', `app-icon-${size}`]

    if (noVerticalAlignment) classes.push('app-va-0')
    if (className) classes.push(className)

    return classes.join(' ')
  }

  return <img src={src as string} alt={alt} className={getClasses()} />
}

export default Icon
