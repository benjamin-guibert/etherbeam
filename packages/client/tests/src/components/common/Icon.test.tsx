import * as React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import Icon from 'components/common/Icon'

describe('<Icon />', () => {
  let component: ReactWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = mount(<Icon src="source" alt="alt" size="xl" className="app-class" />)
    })

    it('should render', () => {
      const img = component.find('img')
      expect(img.hasClass('app-icon')).toBeTruthy()
      expect(img.hasClass('app-icon-xl')).toBeTruthy()
      expect(img.hasClass('app-class')).toBeTruthy()
      expect(img.prop('src')).toBe('source')
      expect(img.prop('alt')).toBe('alt')
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('No vertical alignment', () => {
    beforeAll(() => {
      component = mount(<Icon src="source" alt="alt" size="xl" className="app-class" noVerticalAlignment />)
    })

    it('should render', () => {
      const img = component.find('img')
      expect(img.hasClass('app-icon')).toBeTruthy()
      expect(img.hasClass('app-icon-xl')).toBeTruthy()
      expect(img.hasClass('app-va-0')).toBeTruthy()
      expect(img.hasClass('app-class')).toBeTruthy()
      expect(img.prop('src')).toBe('source')
      expect(img.prop('alt')).toBe('alt')
    })
  })
})
