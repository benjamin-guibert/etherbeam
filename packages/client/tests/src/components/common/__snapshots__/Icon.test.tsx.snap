// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Default should match snapshot 1`] = `
<img
  alt="alt"
  class="app-icon app-icon-xl app-class"
  src="source"
/>
`;
